packages/react-reconciler/src/__tests__/ReactIncrementalErrorHandling-test.internal.js

```diff
@@ -1805,27 +1868,25 @@ describe('ReactIncrementalErrorHandling', () => {
       root.render(<Oops />);
     });
 
-    await act(async () => {
-      // Schedule a default pri and a low pri update on the root.
-      root.render(<Oops />);
-      React.startTransition(() => {
-        root.render(<AllGood />);
-      });
+    await expect(
+      act(async () => {
+        // Schedule a default pri and a low pri update on the root.
+        root.render(<Oops />);
+        React.startTransition(() => {
+          root.render(<AllGood />);
+        });
 
-      // Render through just the default pri update. The low pri update remains on
-      // the queue.
-      await waitFor(['Everything is fine.']);
+        // Render through the default pri and low pri update.
+        await waitFor(['Everything is fine.']);
 
-      // Schedule a discrete update on a child that triggers an error.
-      // The root should capture this error. But since there's still a pending
-      // update on the root, the error should be suppressed.
-      ReactNoop.discreteUpdates(() => {
-        setShouldThrow(true);
-      });
-    });
-    // Should render the final state without throwing the error.
-    assertLog(['Everything is fine.']);
-    expect(root).toMatchRenderedOutput('Everything is fine.');
+        // Schedule a discrete update on a child that triggers an error.
+        ReactNoop.discreteUpdates(() => {
+          setShouldThrow(true);
+        });
+      }),
+    ).rejects.toThrow('Oops');
+    assertLog([]);
+    expect(root).toMatchRenderedOutput(null);
```

Comment:
Needs discussion. Might be fine since we already rendered the low-pri update before we error. So last render wins?